app/views/media/_basics.html.erb

```diff
@@ -54,30 +54,15 @@
       <%= f.label :editor_ids,
                   t('basics.editors') %>
       <%= helpdesk(t('admin.medium.info.editors'), true) %>
-      <% if current_user.admin || current_user.editor? %>
-        <%= f.select :editor_ids,
-                     options_for_select(add_prompt(medium.editors
-                                                         .map { |e| [e.info, e.id] }),
-                                        medium.editors.map(&:id)),
-                     {},
-                     { multiple: true,
-                       class: 'selectize',
-                       style: 'display: none;',
-                       data: { ajax: true,
-                               filled: false,
-                               model: 'user',
-                               placeholder: t('basics.enter_two_letters'),
-                               no_results: t('basics.no_results') } } %>
-      <% else %>
-        <%= f.select :editor_ids,
-                     options_for_select((medium.teachable.speakers | medium.editors).map { |e| [e.info, e.id] },
-                                        medium.editors.map(&:id)),
-                     {},
-                     { multiple: true,
-                       class: 'selectize',
-                       style: 'display: none;',
-                       data: { ajax: false } } %>
-      <% end %>
+      <%= f.select :editor_ids,
+                   options_for_select(medium.eligible_editors(current_user)
+                                            .map { |e| [e.info, e.id] },
+                                      medium.editors.map(&:id)),
+                   {},
+                   { multiple: true,
+                     class: 'selectize',
+                     style: 'display: none;',
+                     data: { ajax: false } } %>
```

Comment:
Generally we have the philosophy (at least until now) to use ajax only when we are dealing with expected larger datasets and retrieve the data directly in Rails otherwise. In this case, the number of eligible editors for a single medium is expected to be rather small (in our production setting, I guess the maximum number will be in the low two digits). That's why the form is already completely populated in the Rails view. If we wanted to use ajax, we would have to write a method similar to fill_user_select but which returns only eligible editors instead of all users.

Reply:
ah ok, thanks for clarifying